src/styles/deserializers.ts

```diff
@@ -0,0 +1,60 @@
+/* @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {IdentNode, parseExpressions} from './parsers.js';
+
+
+/**
+ * For our purposes, an enumeration is a fixed set of CSS-expression-compatible
+ * names. When serialized, a selected subset of the members may be specified as
+ * whitespace-separated strings. An enumeration deserializer is a function that
+ * parses a serialized subset of an enumeration and returns any members that are
+ * found as a Set.
+ *
+ * The following example will produce a deserializer for the days of the
+ * week:
+ *
+ * const deserializeDaysOfTheWeek = enumerationDeserializer([
+ *   'Monday',
+ *   'Tuesday',
+ *   'Wednesday',
+ *   'Thursday',
+ *   'Friday',
+ *   'Saturday',
+ *   'Sunday'
+ * ]);
+ */
+export const enumerationDeserializer = <T extends string>(allowedNames: T[]) =>
```

Comment:
Topic for another day perhaps, but I notice this whole file is only used in one place, for distinguishing which quickLookBrowser we're willing to use. I'm not entirely sure why we do that? Curious if we have an opportunity to remove some code at some point...

Reply:
Yah, this seems ripe for removal or refactor, but it's not obvious what to do and seemed orthogonal to this change.